### ISSUE TITLE
Convert lora models of higher rank into lower rank

### ISSUE THREAD

**Issue by wsxiaoys** [cloneofsimo/lora#142]
For unknown reason, anime community seems to always using R=128 as rank size in their lora fine tunning. Which I [highly] doubt is ineffective.

One experiment might help is to simply use SVD to lower the rank and visualize the result to compare.

**Comment by brian6091** [comment]
You mean to post-process a model? That would be straightforward to do (there's some experiments in Adobe's Custom Diffusion paper on post-tuning decomposition that might interest you).

Rank 128 seems like a lot, not sure you need that high. I find ~30 pretty good for face details. 

Sounds like a good experiment!

**Comment by cloneofsimo** [comment]
Yeah definitely doable experiment. Just for the reference lowering the rank from fully trained model #140

**Comment by ExponentialML** [comment]
From my testing, Tthere's also a lot of variance in how the model is trained that needs to be taken into account.

When you use lower LoRA ranks, you generally need to tune the parameters pretty high (LR: 3e-4), whereas with higher ranks it's plausible to use lower learning rates where it wouldn't learn before (LR: 5e-6).

It makes it kind of difficult to test results when changing the LoRA rank drastically since the hyperparameters are very sensitive to what rank you use.
